CHANGELOG.md

```diff
@@ -11,17 +11,36 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - Run name printed at the end of a run so it's easier to find.
 - Type information added to package data. See [PEP 561](https://www.python.org/dev/peps/pep-0561) for more information.
+- A new integration, `transformers`, with two new steps for running seq2seq models.
+- Added `logging_tqdm`, if you don't want a progress bar, but you still want to see progress in the logs.
+- Added `threaded_generator()`, for wrapping generators so that they run in a separate thread from the generator's consumer.
+- Added a new example for evaluating the T0 model on XSum, a summarization task.
+- Added `MappedSequence` for functionally wrapping sequences.
+- Added `TextFormat`, in case you want to store the output of your steps in raw text instead of JSON.
+- Steps can now list arguments in `SKIP_ID_ARGUMENTS` to indicate that the argument should not affect a step's
+  unique id. This is useful for arguments that affect the execution of a step, but not the output.
+- `Step` now implements `__str__`, so steps look pretty in the debugger.
+- Added `DatasetCombineStep`, a step that combines multiple datasets into one.
 
 ### Fixed
 
 - Fixed torch `StepEarlyCallback` state not being recovered properly on restarts.
+- `LocalWorkspace` now works properly with uncacheable steps.
+- When a Tango run got killed hard, with `kill -9`, or because the machine lost power, `LocalWorkspace` would
+  sometimes keep a step marked as "running", preventing further executions. This still happens sometimes, but it
+  is now much less likely (and Tango gives you instructions for how to fix it).
+- To make all this happen, `LocalWorkspace` now saves step info in a Sqlite database. Unfortunately that means that
+  the workspace format changes and existing workspace directories won't work properly with it.
```

Comment:
Oof, this is not ideal for Zhaofeng's project if we ever want to upgrade tango. Is it possible to write a script that converts an old workspace into a new one?

Reply:
I will look into making it compatible.

Reply:
2cd8994a99c9d2b228ed776186946d6bd3bf29d0

Reply:
Yay!

Reply:
That ended up way bigger than intended because it turns out we had no tests for local workspaces (😳), and to write those tests I had to fix another issue that I wanted to leave for another PR.